### ISSUE TITLE
`MatchingDeclarationName` should have option to exclude

### ISSUE THREAD

**Issue by GrzegorzBobryk** [detekt/detekt#7639]
## Expected Behavior of the rule

In rule [MatchingDeclarationName](https://detekt.dev/docs/rules/naming/#matchingdeclarationname) I want to have possibility to ignore files that contains some string in name. I want to set it in config file.

## Context

In our project we have MVI structure. We have agreed to keep all classes and interfaces for it in one file for each screen. This files have `Contract` suffix in name. Sometime if screen is "dumb" it could have only one class inside of contract. We want to use `MatchingDeclarationName` in our project without putting suppression manually in every such file.

**Comment by cortinico** [comment]
I think you should be able to achieve this with the `excludes` directive at the rule level:
https://github.com/detekt/detekt/blob/f85f4f8f8830cdbdeda6ac65a7d3bc12cbc66374/detekt-core/src/main/resources/default-detekt-config.yml#L333-L336

**Comment by GrzegorzBobryk** [comment]
The contract files are stored in the UI packages with screens and view models. I cannot use your proposition.

**Comment by BraisGabin** [comment]
Why not? You can use glob to match those strings. Otherwise I recommend you to create your own custom rule to implement your use case because it sounds quite unique.

**Comment by cortinico** [comment]
> This files have `Contract` suffix in name

Specifically because of this you can do stuff like:

```yaml
   excludes: ['**/*Contract.kt'] 
```

**Comment by GrzegorzBobryk** [comment]
You are right! Thank you :) I used this incorrectly before. Sorry for the inconvenience.
